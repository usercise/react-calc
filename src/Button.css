.Button {
  border-radius: 6px;
  padding: 4px;
}
