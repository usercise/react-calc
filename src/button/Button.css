.Button {
  border-width: 0;
  border-radius: 6px;
  padding: 4px;
  background-color: rgb(221, 146, 34);
  color: #fff;
  line-height: 1.5em;
  font-size: 1.2rem;
  font-weight: 600;
}

.Button.isActive {
  color: rgb(221, 146, 34);
  background-color: #fff;
}

.Button--plus {
  grid-row: 2 / 4;
  grid-column: 4;
}

.Button--equal {
  grid-row: 4 / 6;
  grid-column: 4;
}

.Button--zero {
  grid-column: 1 / 3;
}
