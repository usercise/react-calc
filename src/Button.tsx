import React from 'react';
import './Button.css';

interface ButtonProps {
  title: string;
  callback: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  buttonValue?: any;
  modifier?: string;
}

const Button: React.SFC<ButtonProps> = props => {
  const classNames = props.modifier
    ? `Button Button--${props.modifier}`
    : 'Button';
  return (
    <button
      className={classNames}
      type="button"
      value={props.buttonValue}
      onClick={props.callback}
    >
      {props.title}
    </button>
  );
};

export default Button;
