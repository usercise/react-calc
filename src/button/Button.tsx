import React from 'react';
import './Button.css';

interface ButtonProps {
  title: string;
  callback: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  buttonValue?: any;
  modifier?: string;
  isActive?: boolean;
}

const Button: React.SFC<ButtonProps> = props => {
  const isActive = props.isActive ? ' isActive' : '';
  const classNames = props.modifier
    ? `Button${isActive} Button--${props.modifier}`
    : `Button${isActive}`;
  return (
    <button
      className={classNames}
      type="button"
      value={props.buttonValue}
      onClick={props.callback}
    >
      {props.title}
    </button>
  );
};

export default Button;
