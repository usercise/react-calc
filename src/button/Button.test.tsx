import React from 'react';
import { shallow, render } from 'enzyme';
import Button from './Button';

test('renders without crashing', () => {
  expect(shallow(<Button title="test" callback={() => {}} />).text()).toEqual(
    'test'
  );
});

test('title gets set', () => {
  expect(shallow(<Button title="test" callback={() => {}} />).text()).toEqual(
    'test'
  );

  expect(
    shallow(<Button title="testy" callback={() => {}} />).text()
  ).not.toEqual('test');
});

test('Button to not have changed since test', () => {
  const component = render(<Button title="test" callback={() => {}} />);
  let tree = component.serialize();
  expect(tree).toMatchSnapshot();
});

test('class name to match element', () => {
  expect(
    shallow(<Button title="test" callback={() => {}} />).hasClass('Button')
  ).toBeTruthy();
});

test('modifier gets set', () => {
  expect(
    shallow(
      <Button title="test" callback={() => {}} modifier="test" />
    ).hasClass('Button--test')
  ).toBeTruthy();
});
