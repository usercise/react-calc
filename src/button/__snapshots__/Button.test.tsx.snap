// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to not have changed since test 1`] = `""`;
